com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/mqtt_streams.py

```diff
@@ -0,0 +1,208 @@
+ #      </td></tr>
+ # <tr><td>password</td>
+ #      <td>Optional String.  The identifier to use when authenticating
+ #      with server configured to require that form of authentication.
+ #      </td></tr>
+ # <tr><td>trustStore</td>
+ #      <td>Optional String. The pathname to a file containing the
+ #      public certificate of trusted MQTT servers.  If a relative path
+ #      is specified, the path is relative to the application directory.
+ #      Required when connecting to a MQTT server with an 
+ #      ssl:/... serverURI.
+ #      </td></tr>
+ # <tr><td>trustStorePassword</td>
+ #      <td>Required String when {@code trustStore} is used.
+ #      The password needed to access the encrypted trustStore file.
+ #      </td></tr>
+ # <tr><td>keyStore</td>
+ #      <td>Optional String. The pathname to a file containing the
+ #      MQTT client's public private key certificates.
+ #      If a relative path is specified, the path is relative to the
+ #      application directory. 
+ #      Required when an MQTT server is configured to use SSL client authentication.
+ #      </td></tr>
+ # <tr><td>keyStorePassword</td>
+ #      <td>Required String when {@code keyStore} is used.
+ #      The password needed to access the encrypted keyStore file.
+ #      </td></tr>
+ # <tr><td>receiveBufferSize</td>
+ #      <td>[subscribe] Optional Integer. The size, in number
+ #      of messages, of the subscriber's internal receive buffer.  Received
+ #      messages are added to the buffer prior to being converted to a
+ #      stream tuple. The receiver blocks when the buffer is full.
+ #      The default is 50.
+ #      </td></tr>
+ # <tr><td>retain</td>
+ #      <td>[publish] Optional Boolean. Indicates if messages should be
+ #      retained on the MQTT server.  Default is false.
+ #      </td></tr>
+ # <tr><td>defaultQOS</td>
+ #      <td>Optional Integer. The default
+ #      MQTT quality of service used for message handling.
+ #      The default is 0.
+ #      </td></tr>
+ # </table>
+ # 
+ # @see <a href="http://mqtt.org">http://mqtt.org</a>
+ # @see <a
+ #      href="http://ibmstreams.github.io/streamsx.messaging/">com.ibm.streamsx.messaging</a>
+
+
+    def __init__(self,topology,config) :
+        self.topology = topology
+        self.config = config.copy()
+        self.opCnt = 0
+        
+    def publish(self,src_op,topic=None) :
+        if topic is None:
+            raise TypeError("No topic provided to MqttStreams.publish")
+        parms = self.config.copy()
+        #parms['reconnectionBound'] = int(-1) (needs to be int vs long)
+        #parms['qos'] = int(0) (needs to be int vs long)
+        #del parms['messageQueueSize']
+        if topic is None :
+            parms['topicOutAttrName'] = "topic"
+        else :
+            parms['topic'] = topic
+        parms['dataAttributeName'] = "message"
+        if (++self.opCnt > 1) :
+            # each op requires its own clientID
+            clientId =  params['clientID']
```

Comment:
I will change the code to: add the python object id to ensure uniqueness
params['clientID'] = id()+"-"+opCnt+"-"+clientId